apps/files/js/filelist.js

```diff
@@ -1787,8 +1787,11 @@
 			td.append(linkElem);
 			tr.append(td);
 
-			var enabledThemes = window.OCA?.Theming?.enabledThemes || []
-			var isDarkTheme = enabledThemes.join('').indexOf('dark') !== -1
+			const enabledThemes = window.OCA?.Theming?.enabledThemes || []
+			// Check enabled themes, if system default is selected check the browser
+			const isDarkTheme = (enabledThemes.length === 0 || enabledThemes[0] === 'default')
+				? window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches
```

Comment:
```suggestion
				? window.matchMedia('(prefers-color-scheme: dark)').matches
```
Would this work @susnux?

Reply:
As long as the browser supports the [`matchMedia` API](https://caniuse.com/?search=matchMedia): yes

Reply:
Could we remove the check since it's seems safe going by our browser target rules https://github.com/nextcloud/browserslist-config/blob/v2.2.0/browserlist.config.js and the compatible browsers https://browserslist.dev/?q=PjAuMjUlLCBub3QgaWUgMTEsIG5vdCBvcF9taW5pIGFsbCwgbm90IGRlYWQsIEZpcmVmb3ggRVNS?

Reply:
Done :)